include/SDL3/SDL_gpu.h

```diff
@@ -941,212 +941,373 @@ typedef enum SDL_GPUDriver
+ *
+ * \sa SDL_CopyGPUBufferToBuffer
+ */
 typedef struct SDL_GPUBufferLocation
 {
-    SDL_GPUBuffer *buffer;
-    Uint32 offset;
+    SDL_GPUBuffer *buffer;  /**< The buffer. */
+    Uint32 offset;          /**< The starting byte within the buffer. */
 } SDL_GPUBufferLocation;
 
+/**
+ * A structure specifying a region of a buffer.
+ *
+ * Used when transferring data to or from buffers.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_UploadToGPUBuffer
+ * \sa SDL_DownloadFromGPUBuffer
+ */
 typedef struct SDL_GPUBufferRegion
 {
-    SDL_GPUBuffer *buffer;
-    Uint32 offset;
-    Uint32 size;
+    SDL_GPUBuffer *buffer;  /**< The buffer. */
+    Uint32 offset;          /**< The starting byte within the buffer. */
+    Uint32 size;            /**< The size in bytes of the region. */
 } SDL_GPUBufferRegion;
 
-/* Note that the `first_vertex` and `first_instance` parameters are NOT compatible with
+/**
+ * A structure specifying the parameters of an indirect draw command.
+ *
+ * Note that the `first_vertex` and `first_instance` parameters are NOT compatible with
  * built-in vertex/instance ID variables in shaders (for example, SV_VertexID). If
  * your shader depends on these variables, the correlating draw call parameter MUST
  * be 0.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_DrawGPUPrimitivesIndirect
  */
 typedef struct SDL_GPUIndirectDrawCommand
 {
-    Uint32 num_vertices;   /* number of vertices to draw */
-    Uint32 num_instances;  /* number of instances to draw */
-    Uint32 first_vertex;   /* index of the first vertex to draw */
-    Uint32 first_instance; /* ID of the first instance to draw */
+    Uint32 num_vertices;   /**< The number of vertices to draw. */
+    Uint32 num_instances;  /**< The number of instances to draw. */
+    Uint32 first_vertex;   /**< The index of the first vertex to draw. */
+    Uint32 first_instance; /**< The ID of the first instance to draw. */
 } SDL_GPUIndirectDrawCommand;
 
+/**
+ * A structure specifying the parameters of an indexed indirect draw command.
+ *
+ * Note that the `first_vertex` and `first_instance` parameters are NOT compatible with
+ * built-in vertex/instance ID variables in shaders (for example, SV_VertexID). If
+ * your shader depends on these variables, the correlating draw call parameter MUST
+ * be 0.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_DrawGPUIndexedPrimitivesIndirect
+ */
 typedef struct SDL_GPUIndexedIndirectDrawCommand
 {
-    Uint32 num_indices;    /* number of vertices to draw per instance */
-    Uint32 num_instances;  /* number of instances to draw */
-    Uint32 first_index;    /* base index within the index buffer */
-    Sint32 vertex_offset;  /* value added to vertex index before indexing into the vertex buffer */
-    Uint32 first_instance; /* ID of the first instance to draw */
+    Uint32 num_indices;    /**< The number of vertices to draw per instance. */
```

Comment:
It’s slightly confusing that the parameter is num_indices but it represents the number of vertices. Should we rename the parameter?

Reply:
The comment is wrong, it is the number of indices.